pkg/backup/backup.go

```diff
@@ -443,9 +447,60 @@ func (kb *kubernetesBackupper) BackupWithResolvers(
 			Name:          items[i].name,
 		}
 		itemsMap[key] = append(itemsMap[key], items[i])
+		// add to total items for progress reporting
+		if items[i].kind != "" {
+			backupRequest.BackedUpItems.AddItemToTotal(itemKey{
+				resource:  fmt.Sprintf("%s/%s", items[i].preferredGVR.GroupVersion().String(), items[i].kind),
+				namespace: items[i].namespace,
+				name:      items[i].name,
+			})
+		}
 	}
 
 	var itemBlock *BackupItemBlock
+	itemBlockReturn := make(chan ItemBlockReturn, max(backupRequest.ItemBlockWorkerCount, 10))
```

Comment:
If this go routine of receiving from `itemBlockReturn` is blocked or simply slower than the workers processing itemBlocks, for any reason, this `itemBlockReturn` channel will be a bottleneck, even though we make it as `backupRequest.ItemBlockWorkerCount`.

So I think we need to make the channel size for `itemBlockReturn` larger, excessive large is not a problem. We just need to make sure the itemBlock backupper (the producer) go through any time.

The original code for having `itemBlockReturn` as a no buffered channel is for sure not correct either, but I missed it at the first time.